<a href="https://colab.research.google.com/github/AleOi/PCS3438Eduardo/blob/master/Lista1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lista de Exercícios Inteligência Artificial

In [0]:
import pandas as pd
import matplotlib.pyplot  as plt
import os 

from google.colab import drive

drive.mount('/content/drive')


Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/drive


In [0]:
# Leitura dos arquivos entregues
class01 = pd.read_csv('/content/drive/My Drive/Colab Notebooks/PCS/Classificador/class01.csv')
class02 = pd.read_csv('/content/drive/My Drive/Colab Notebooks/PCS/Classificador/class02.csv')

reg01 = pd.read_csv('/content/drive/My Drive/Colab Notebooks/PCS/Classificador/reg01.csv')
reg02 = pd.read_csv('/content/drive/My Drive/Colab Notebooks/PCS/Classificador/reg02.csv')

In [0]:
class01 = class01.dropna()
class01.isna().sum()
class01.head()

x0        x1        x2        x3  ...       x97       x98       x99  target
0  1.162990 -0.831971 -0.185266  0.441240  ... -0.450784 -0.187694 -0.538329     1.0
1 -1.750976 -0.516614 -0.761588  1.692347  ... -0.066449 -0.802430  1.587229     0.0
2  0.304635  0.972369  0.202908 -0.607761  ... -0.003495 -0.149940  0.539892     8.0
3 -1.667576 -0.262639 -1.513473 -1.060125  ... -0.450510 -0.757631 -3.184493     7.0
4 -0.820670  0.390631  0.761628 -0.945407  ...  1.023761 -0.335391  2.448868     3.0

[5 rows x 101 columns]

In [0]:
from sklearn.model_selection import train_test_split
import time
class01Att = class01.iloc[:, 0:class01.columns.size - 1]
class01Res = class01.iloc[:, class01.columns.size - 1]

X_train, X_test, Y_train, Y_test = train_test_split(class01Att, class01Res, test_size=0.35, random_state=int(time.time()))

In [0]:
X_train.shape

(650, 100)

In [0]:
X_test.shape

(350, 100)

In [0]:
Y_train.shape

(650,)

In [0]:
Y_test.shape

(350,)

Considerando os dados presentes no arquivo01.csv, treine o algoritmo Naive Bayes Gaussiano utilizando a netodologia de validação cruzada houdout (utilize para treino as 350 primeiras linhas e para validação as demais). Qual o valor da acurácia a base de treino? Qual o valor da acurácia a base de validação?

In [0]:
from sklearn.naive_bayes import GaussianNB

gnb = GaussianNB()
# Train
gnb.fit(X_train, Y_train)
y_pred = gnb.predict(X_test)

# Print results
print("Numero de acertos do total {} points : {}, performance {:05.2f}%"
      .format(
          X_test.shape[0],
          (Y_test == y_pred).sum(),
          100*(Y_test == y_pred).sum()/X_test.shape[0]
))

Numero de acertos do total 350 points : 224, performance 64.00%
